python/ml4ir/applications/ranking/tests/test_ranking_kfold_cv.py

```diff
@@ -0,0 +1,170 @@
+        pd.concat([dfs[1][i][-1] for i in range(len(dfs[1]))]).to_csv(working_dir / 'validation' / 'data.csv')
+        pd.concat([dfs[2][i][-1] for i in range(len(dfs[2]))]).to_csv(working_dir / 'test' / 'data.csv')
+        return create_parse_args(data_dir=working_dir.as_posix(),
+                                 feature_config=feature_config_file.as_posix(),
+                                 model_config=model_config_file.as_posix(),
+                                 logs_dir=log_dir.as_posix(),
+                                 num_folds=num_folds,
+                                 use_testset_in_folds=use_testset_in_folds)
+
+    def run_merge_datasets_test(self, dataset_name, num_features, num_folds, use_testset_in_folds, expected_num_queries):
+        args = self.setup_data(dataset_name, num_features, num_folds, use_testset_in_folds)
+
+        rp = RankingPipeline(args=args)
+        relevance_dataset = rp.get_kfold_relevance_dataset(args.kfold, args.include_testset_in_kfold,
+                                                           read_data_sets=True)
+
+        all_data = relevance_dataset.merge_datasets(args.include_testset_in_kfold)
+        query_ids = set([q[0]['query_id'].numpy()[0] for q in all_data])
+        assert len(query_ids) == expected_num_queries
+
+    def run_folds_creation_test(self, dataset_name, num_features, num_folds, use_testset_in_folds):
+        args = self.setup_data(dataset_name, num_features, num_folds, use_testset_in_folds)
+
+        rp = RankingPipeline(args=args)
+        relevance_dataset = rp.get_kfold_relevance_dataset(args.kfold, args.include_testset_in_kfold,
+                                                           read_data_sets=True)
+
+        all_data = relevance_dataset.merge_datasets(args.include_testset_in_kfold)
+        for i in range(num_folds):
+            fold_relevance_dataset = rp.get_kfold_relevance_dataset(args.kfold, args.include_testset_in_kfold,
+                                                               read_data_sets=False)
+            fold_relevance_dataset.create_folds(i, all_data)
+            train = fold_relevance_dataset.train.unbatch()
+            validation = fold_relevance_dataset.validation.unbatch()
+            if use_testset_in_folds:
+                test = fold_relevance_dataset.test.unbatch()
+            train_qids = set([q[0]['query_id'].numpy()[0] for q in train])
+            validation_qids = set([q[0]['query_id'].numpy()[0] for q in validation])
+            if use_testset_in_folds:
+                test_qids = set([q[0]['query_id'].numpy()[0] for q in test])
+            if use_testset_in_folds:
+                assert len(set.intersection(train_qids, test_qids)) == 0
+                assert len(set.intersection(validation_qids, test_qids)) == 0
+            assert len(set.intersection(train_qids, validation_qids)) == 0
+
+    def test_merge_datasets_1(self):
```

Comment:
The test names aren't descriptive. At the very least, we need a doc string explaining what this function is trying to test